From 85f5a57b9bdd94fa7a20280c31f4f8b148368215 Mon Sep 17 00:00:00 2001
From: Andrea Adami <andrea.adami@gmail.com>
Date: Thu, 7 Nov 2013 23:00:31 +0100
Subject: [PATCH] ARM: SA1100: resize collie mtd to SZ_16M

The end-user models have all 2x8 = 16 Megabytes of NOR.
It seems that only some dev models/prototype had 32M.
With current code, giving size of 32M can confuse the detection.

JEDEC:
sa1100-0: Found 2 x16 devices at 0x0 in 32-bit bank
sa1100-0: Found different chip or no chip at all (mfr 0x4, id 0x3001) at 0x1000000
SA1100 flash: CFI device at 0x00000000, 16MiB, 32-bit

CFI:
sa1100-0: Found 2 x16 devices at 0x0 in 32-bit bank. Manufacturer ID 0x0000b0 Chip ID 0x0000b0
sa1100-0: Found 2 x16 devices at 0x1000000 in 32-bit bank
Intel/Sharp Extended Query Table at 0x0039
...
sa1100-0: 2 set(s) of 2 interleaved chips --> 2 partitions of 16384 KiB
SA1100 flash: CFI device at 0x00000000, 32MiB, 32-bit


Signed-off-by: Andrea Adami <andrea.adami@gmail.com>
---
 arch/arm/mach-sa1100/collie.c             |      2 +-
 2 files changed, 230404 insertions(+), 1604 deletions(-)

diff --git a/arch/arm/mach-sa1100/collie.c b/arch/arm/mach-sa1100/collie.c
index 7fb96eb..c1005ef 100644
--- a/arch/arm/mach-sa1100/collie.c
+++ b/arch/arm/mach-sa1100/collie.c
@@ -298,7 +298,7 @@ static struct flash_platform_data collie_flash_data = {
 };
 
 static struct resource collie_flash_resources[] = {
-	DEFINE_RES_MEM(SA1100_CS0_PHYS, SZ_32M),
+	DEFINE_RES_MEM(SA1100_CS0_PHYS, SZ_16M),
 };
 
 static struct sa1100fb_mach_info collie_lcd_info = {
